boot/bootutil/include/bootutil/crypto/hmac_sha256.h

```diff
@@ -76,6 +85,48 @@ static inline int bootutil_hmac_sha256_finish(bootutil_hmac_sha256_context *ctx,
 }
 #endif /* MCUBOOT_USE_TINYCRYPT */
 
+#if defined(MCUBOOT_USE_MBED_TLS)
+/**
+ * The generic message-digest context.
+ */
+typedef mbedtls_md_context_t bootutil_hmac_sha256_context;
+
+static inline void bootutil_hmac_sha256_init(bootutil_hmac_sha256_context *ctx)
+{
+    mbedtls_md_init(ctx);
+}
+
+static inline void bootutil_hmac_sha256_drop(bootutil_hmac_sha256_context *ctx)
+{
+    mbedtls_md_free(ctx);
+}
+
+static inline int bootutil_hmac_sha256_set_key(bootutil_hmac_sha256_context *ctx, const uint8_t *key, unsigned int key_size)
+{
+    int rc;
+
+    rc = mbedtls_md_setup(ctx, mbedtls_md_info_from_string("SHA256"), 1);
+    if (rc == 0) {
+        rc = mbedtls_md_hmac_starts(ctx, key, key_size);
+    }
```

Comment:
One final nitpick would be to change this block from checking success to the "ïf fail, return early" typical pattern used through-out the code.

Reply:
I prefer to do not use early or multiple returns in a functions as it is against MISRA.

Will fix if it is essential for the project.

Reply:
I run a smoke test on an actual HW each time before commit.

mbedTLS which is a platform-agnostic SW library so should work in any case.

Reply:
> I prefer to do not use early or multiple returns in a functions as it is against MISRA.
> 
> Will fix if it is essential for the project.

I honestly did not know, sad. Anyway good taste would implicate in using the same style that is already employed, not whatever style one came up with because of this or that, but I'll leave it at your discression.

> I run a smoke test on an actual HW each time before commit.
> 
> mbedTLS is a platform-agnostic SW library so should work in any case.

Well, I don't want to sit here and test it every time someone changes the code, also I don't want you to do it, so I guess my question was a bit of a tricky question for which your correct answer would have been something like: "Ha, I got this neat idea, why don't we run it on the simulator so we know it's always working?", for which I would answer: "Awesome idea! Add a new commit doing this!".

Reply:
> I prefer to do not use early or multiple returns in a functions as it is against MISRA.

It's not actually against MISRA. Not having early returns, and a goto to the end of the function are only recommendations, not requirements (for example, a requirement is to not have gotos to arbitrary parts of the code).

Computer science has learned quite a bit since 1998. One of those things is that early return, or a goto for cleanup results in cleaner, and more understandable code than code that uses what ends up being a convoluted set of if statements to try and work around that kind of restriction. There are numerous projects that follow this, including the Linux kernel, and even the Zephyr project (which seeks MIRSA compliance). Several newer programming languages (Rust and Go, specifically) encourage the use of early returns to make for clearer code.

So, yes, please use early return, or a goto to cleanup code if needed. It makes the code easier to understand, and will not be any kind of hindrance to MISRA compliance, if desired.

Reply:
did not have an intent to argue or insist, just explained the motivation

surely, updated to match overall style